src/types/mutex.rs

```diff
@@ -0,0 +1,98 @@
+use std::cell::UnsafeCell;
+use std::ops::{Deref, DerefMut};
+
+/// Wrapper for [`PyMutex`](https://docs.python.org/3/c-api/init.html#c.PyMutex), exposing an RAII guard interface similar to `std::sync::Mutex`.
+#[derive(Debug)]
+pub struct PyMutex<T> {
+    _mutex: UnsafeCell<crate::ffi::PyMutex>,
```

Comment:
```suggestion
    inner: UnsafeCell<crate::ffi::PyMutex>,
```

To avoid a bunch of `self.mutex_mutex` or `self.mutex.mutex` (if the underscore were to be removed)